### PR TITLE
[MNEDC] Changed config to YAML format

### LINKED ISSUE #209: [Folder Structure] Follow standard Go project Layout
There is a popular [**standard** Go application project layout](https://github.com/golang-standards/project-layout) and many projects follow it.
I think we should change the folder structure in terms of this guide like below.
- `GoMain/src/main` > `cmd`
- `Buillder` > `build`
- `samples` > `examples`
- `samples/datastorage` > `configs/datastorage`
- `src` > `internal`
- `doc` > `docs`
- `doc/edge_orchestration_api.yaml` and `doc/edge_orchestration_api_secure.yaml` > `api/...`

Please add or rectify the list above since it might be incorrect!

### PR DESCRIPTION
**Pull request by sun-sharma**
- To adopt proper go styles and format transformed the client.config file to YAML file (Making it more human readable)
- Made changes in client side for reading the YAML file to get the server's actual IP and port

Signed-off-by: Sunchit Sharma <sun.sharma@samsung.com>

# Description

client.config for MNEDC client has been changed to YAML format. This has been done for maintaining proper format , styling and making it more human readable. Code changes have been done to parse and read the files. Tests have also been accordingly modified.

Fixes #209 

## Type of change

- [X] Code cleanup/refactoring
- [X] Documentation update

# How Has This Been Tested?

1. Set Up one system in Main NAT as MNEDC server
2. Set up 2 system in Sub NAT as MNEDC Client
3. Checked the logs for connection
4. Requested Service from Sub NAT system to Main NAT system
5. Worked Properly as expected.

## Logs of client machine post change 

```

INFO[2021-03-04T10:49:33Z]internalhandler.go:344 APIV1DiscoverymgrMNEDCDeviceInfoPost RestInternalInterface Private Add: 192.168.1.29 

INFO[2021-03-04T10:49:33Z]internalhandler.go:345 APIV1DiscoverymgrMNEDCDeviceInfoPost RestInternalInterface Virtual Add: 10.0.0.1

```

**Test Configuration**:
* Firmware version:  Ubuntu 16.04
* Hardware:  x86-64
* Edge Orchestration Release:  Coconut

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: internal/controller/discoverymgr/discovery.go [hunk 1]
```diff
@@ -859,3 +851,23 @@ func activeDiscovery() {
 func resetServer(ips []net.IP) {
 	wrapperIns.ResetServer(ips)
 }
+
+// getMNEDCServerAddress fetches the IP and Port of the server
+func getMNEDCServerAddress(path string) (string, string, error) {
+
+	var serverIP, serverPort string
```
Comment (t25kim) [inline internal/controller/discoverymgr/discovery.go:858]: You can implement the function without these variables.

File: internal/controller/discoverymgr/mnedc/client/client.go [hunk 2]
```diff
@@ -482,22 +481,22 @@ func (c *Client) SetClient(clientAPI restclient.Clienter) {
 	c.clientAPI = clientAPI
 }
 
+// getMNEDCServerAddress fetches the IP and Port of the server
 func getMNEDCServerAddress(path string) (string, string, error) {
 
 	var serverIP, serverPort string
```
Comment (t25kim) [inline internal/controller/discoverymgr/mnedc/client/client.go:487]: You can remove this line.

**Review by t25kim — COMMENTED** [review]
Thank you for your contribution!
Please update the [test-suite.yml](https://github.com/lf-edge/edge-home-orchestration-go/blob/bb49416e134e68696a9c8d27e58c69620b330de6/.github/workflows/test-suite.yml#L24) as well and take a look at my comments.

File: configs/mnedc/client-config.yaml [hunk 3]
```diff
@@ -0,0 +1,3 @@
+# This configuration file is for providing the Actual IP and Port of the MNEDC server to the clients
```
Comment (t25kim) [inline configs/mnedc/client-config.yaml:1]: Please add a note that `server-ip` and `port` need to be modified for each environment.

**Comment by sun-sharma** [comment]
Hi @t25kim , Thanks for the review. I have made the changes as you suggested.

**Review by t25kim — APPROVED** [review]
LGTM. Thanks!

**Review by tdrozdovsky — APPROVED** [review]
LGTM

**Review by MoonkiHong — APPROVED** [review]
LGTM.

**Pull request merged**